### PR TITLE
feat(rest): add support for ajv-errors

### PR DESCRIPTION
**Pull request by mastermunj**
Implements #2146

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [x] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

**Comment by mastermunj** [comment]
Hi @achrinza, added docs for [Model > Property Decorator](https://github.com/strongloop/loopback-next/blob/master/docs/site/Model.md#property-decorator) and [Parsing Requests > Request Body](https://github.com/strongloop/loopback-next/blob/master/docs/site/Parsing-requests.md#request-body).

File: packages/rest/src/validation/request-body.validator.ts [hunk 1]
```diff
@@ -190,6 +191,7 @@ function createValidator(
     {},
     {
       allErrors: true,
+      jsonPointers: true,
```
Comment (raymondfeng) [inline packages/rest/src/validation/request-body.validator.ts]: Is it needed for ajv-errors?

File: packages/rest/src/__tests__/unit/request-body.validator.test.ts [hunk 2]
```diff
@@ -333,7 +333,7 @@ describe('validateRequestBody', () => {
     it('nested $ref in reference', () => {
       const details: RestHttpErrors.ValidationErrorDetails[] = [
         {
-          path: '.accounts[0].address.city',
+          path: '/accounts/0/address/city',
```
Comment (raymondfeng) [inline packages/rest/src/__tests__/unit/request-body.validator.test.ts]: Is it introduced by `jsonPointers: true,`?

File: packages/rest/src/validation/request-body.validator.ts (hunk 1 above)
Comment (mastermunj) [reply packages/rest/src/validation/request-body.validator.ts]: Yes, `allErrors` & `jsonPointers` both are needed by `ajv-errors` as documented in their [usage docs](https://github.com/epoberezkin/ajv-errors#usage).

When not provided, they're enabled from within `ajv-errors` at https://github.com/epoberezkin/ajv-errors/blob/master/index.js#L4-L8

Benefits of jsonPointers are listed at [ajv errors -> Templates](https://github.com/epoberezkin/ajv-errors#templates)

File: packages/rest/src/__tests__/unit/request-body.validator.test.ts (hunk 2 above)
Comment (mastermunj) [reply packages/rest/src/__tests__/unit/request-body.validator.test.ts]: Yes, it is because of `jsonPointers: true`

**Pull request merged**